### PR TITLE
 RavenDB-22347 Add comprehensive tests for dictionary patching with special character keys

### PR DESCRIPTION
**Pull request by ayende**
### Issue link

  https://issues.hibernatingrhinos.com/issue/RavenDB-22347

  ### Additional description

  Follow-up to PR #22127 (dictionary patching with variable keys and special characters).

  1. **Comprehensive test coverage** — 20 additional tests on top of the PR's original 6, covering:
     - Indexer patch with keys containing quotes, dashes, spaces, dots, backslashes
     - Add/Remove via variable keys and object property keys
     - Patch merging (multiple Add/Remove in same session)
     - KeyValuePair overload with special character keys
     - Escape edge cases: newlines, tabs, mixed special characters, empty string key
     - Enum key regression tests (Add, Remove, Indexer)
     - Complex object values with special character keys
     - Regression tests ensuring non-dictionary Patch and Increment still work after PatchPathWrappedConstantSupport was added to path compilation

  2. **Replace hand-rolled JS string escaping with `JsonConvert.ToString`** — Consolidates three duplicate escape implementations (`EscapeForJsString`, `WriteStringLiteral`, and inline escaping in `FormatKeyForJavaScript`) into a single
  `ToJsStringLiteral` helper. 

  ### Type of change

  - [x] Bug fix

  ### How risky is the change?

  - [x] Low

  ### Backward compatibility

  - [x] Non breaking change

  ### Is it platform specific issue?

  - [x] No

  ### Documentation update

  - [x] No documentation update is needed

  ### Testing by Contributor

  - [x] Tests have been added that prove the fix is effective or that the feature works
  - [x] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`)

  ### Testing by RavenDB QA team

  - [x] No special testing by RavenDB QA team is needed

  ### Is there any existing behavior change of other features due to this change?

  - [x] No

  ### UI work

  - [x] No UI work is needed

### CODE REVIEW

File: test/FastTests/Issues/RavenDB-22347.cs [hunk 1]
```diff
@@ -0,0 +1,645 @@
...
+        [InlineData("line1\nline2")]
+        [InlineData("col1\tcol2")]
+        [InlineData("")]
+        [InlineData("\r\n")]
+        [InlineData("\u2028")]
+        [InlineData("\u2029")]
+        [InlineData("\u3712")]
+        [InlineData("\ud83d\ude00")]
+        [InlineData("key \ud83d\udd11 with emoji")]
+        public void CanPatchDictionaryViaIndexerWithVariousKeys(string key)
```
Comment (Lwiel) [inline test/FastTests/Issues/RavenDB-22347.cs:31]: Can we add `'key'` and `"key"` to tests data?

Comment (Lwiel) [reply test/FastTests/Issues/RavenDB-22347.cs:33]: Added
